applications/blackhole/src/blackhole_socket_callback.erl

```diff
@@ -39,6 +39,29 @@ recv(_SessionPid, _SessionId, {'event', _Ignore, <<"subscribe">>, SubscriptionJO
     end,
     {'ok', Context1};
 
+recv(SessionPid, SessionId, {'event', _Ignore, <<"unsubscribe">>, {[]}}, Context) ->
+    lager:debug("remove all bindings for session: ~p", [SessionId]),
+    Filter = fun (_1, _2, _3, _4) -> filter_bindings(SessionPid, _1, _2, _3, _4) end,
```

Comment:
The underscore suggests these variables will be unused. Since you are obviously using them, use real variable names. `fun(A, B, C, D) -> filter_bindings(SessionPid, A, B, C, D) end` would be fine.